__tests__/unit/loops.test.ts

```diff
@@ -0,0 +1,789 @@
+import * as ts from "typescript";
+import { TranspileError } from "../../src/TranspileError";
+import { LuaLibImportKind, LuaTarget } from "../../src/CompilerOptions";
+import * as util from "../util";
+
+test.each([{ inp: [0, 1, 2, 3], expected: [1, 2, 3, 4] }])("while (%p)", ({ inp, expected }) => {
+    const result = util.transpileAndExecute(
+        `let arrTest = ${JSON.stringify(inp)};
+        let i = 0;
+        while (i < arrTest.length) {
+            arrTest[i] = arrTest[i] + 1;
+            i++;
+        }
+        return JSONStringify(arrTest);`,
+    );
+
+    expect(result).toBe(JSON.stringify(expected));
+});
+
+test.each([{ inp: [0, 1, 2, 3, 4], expected: [0, 1, 2, 1, 4] }])(
+    "while with continue (%p)",
+    ({ inp, expected }) => {
+        const result = util.transpileAndExecute(
+            `let arrTest = ${JSON.stringify(inp)};
+    let i = 0;
+    while (i < arrTest.length) {
+        if (i % 2 == 0) {
+            i++;
+            continue;
+        }
+        let j = 2;
+        while (j > 0) {
+            if (j == 2) {
+                j--
+                continue;
+            }
+            arrTest[i] = j;
+            j--;
+        }
+
+        i++;
+    }
+    return JSONStringify(arrTest);`,
+        );
+
+        expect(result).toBe(JSON.stringify(expected));
+    },
+);
+
+test.each([{ inp: [0, 1, 2, 3, 4], expected: [0, 1, 2, 1, 4] }])(
+    "dowhile with continue (%p)",
+    ({ inp, expected }) => {
+        const result = util.transpileAndExecute(
+            `let arrTest = ${JSON.stringify(inp)};
+    let i = 0;
+    do {
+        if (i % 2 == 0) {
+            i++;
+            continue;
+        }
+        let j = 2;
+        do {
+            if (j == 2) {
+                j--
+                continue;
+            }
+            arrTest[i] = j;
+            j--;
+        } while (j > 0)
+
+        i++;
+    } while (i < arrTest.length)
+    return JSONStringify(arrTest);`,
```

Comment:
Indenting is messed up here

Reply:
I have fixed it in this file, but looks like there are more similar cases. I'll look into solving it with a codemod